rpcs3/Emu/Cell/Modules/cellAudio.cpp

```diff
@@ -966,21 +1001,47 @@ void cell_audio_thread::mix(float* out_buffer, s32 offset)
 				{
 					out_buffer[out + 0] += left;
 					out_buffer[out + 1] += right;
-					out_buffer[out + 2] += center;
-					out_buffer[out + 3] += low_freq;
-					out_buffer[out + 4] += side_left + rear_left;
-					out_buffer[out + 5] += side_right + rear_right;
+
+					if constexpr (_channels >= 6) // Only mix the surround channels into the output if surround output is configured
+					{
+						out_buffer[out + 2] += center;
+						out_buffer[out + 3] += low_freq;
+
+						if constexpr (_channels == 6)
+						{
+							out_buffer[out + 4] += side_left + rear_left;
+							out_buffer[out + 5] += side_right + rear_right;
+						}
+						else // When using 7.1 ouput, surround channels need to be mixed into the last two channels
+						{
+							out_buffer[out + 6] += side_left + rear_left;
+							out_buffer[out + 7] += side_right + rear_right;
+						}
+					}
```

Comment:
this change is unexplained.
Where is this coming from?

Reply:
Rear channels come before Surround/Side channels in 7.1 output.

Reply:
I'll change the comment to be more clear

Reply:
Yeah I assumed as much. you'd hear rear from the sides otherwise.
But the intent was a bit unclear in the code